src/movement/control_system.cpp

```diff
@@ -1,557 +1,573 @@
-    current_state.yaw_left_state =  state_to_string(goal_types[5]);
-
-    current_state.forward_goal = goals[0];
-    current_state.strafe_left_goal = goals[1];
-    current_state.dive_goal = goals[2];
-    current_state.roll_right_goal = goals[3] / _PI_OVER_180;
-    current_state.pitch_down_goal = goals[4] / _PI_OVER_180;
-    current_state.yaw_left_goal = goals[5] / _PI_OVER_180;
-
-    current_state.forward_error = current_error[0];
-    current_state.strafe_left_error = current_error[1];
-    current_state.dive_error = current_error[2];
-    current_state.roll_right_error = current_error[3] / _PI_OVER_180;
-    current_state.pitch_down_error = current_error[4] / _PI_OVER_180;
-    current_state.yaw_left_error = current_error[5] / _PI_OVER_180;
-
-    current_state.forward_integral = current_integral[0];
-    current_state.strafe_left_integral = current_integral[1];
-    current_state.dive_integral = current_integral[2];
-    current_state.roll_right_integral = current_integral[3] / _PI_OVER_180;
-    current_state.pitch_down_integral = current_integral[4] / _PI_OVER_180;
-    current_state.yaw_left_integral = current_integral[5] / _PI_OVER_180;
-
-    current_state.forward_derivative = current_derivative[0];
-    current_state.strafe_left_derivative = current_derivative[1];
-    current_state.dive_derivative = current_derivative[2];
-    current_state.roll_right_derivative = current_derivative[3] / _PI_OVER_180;
-    current_state.pitch_down_derivative = current_derivative[4] / _PI_OVER_180;
-    current_state.yaw_left_derivative = current_derivative[5] / _PI_OVER_180;
-
-    current_state.forward_rotation_control = rotation_control[0];
-    current_state.strafe_left_rotation_control = rotation_control[1];
-    current_state.dive_rotation_control = rotation_control[2];
-    current_state.roll_right_rotation_control = rotation_control[3] / _PI_OVER_180;
-    current_state.pitch_down_rotation_control = rotation_control[4] / _PI_OVER_180;
-    current_state.yaw_left_rotation_control = rotation_control[5] / _PI_OVER_180;
-
-    current_state.forward_translation_control = translation_control[0];
-    current_state.strafe_left_translation_control = translation_control[1];
-    current_state.dive_translation_control = translation_control[2];
-    current_state.roll_right_translation_control = translation_control[3] / _PI_OVER_180;
-    current_state.pitch_down_translation_control = translation_control[4] / _PI_OVER_180;
-    current_state.yaw_left_translation_control = translation_control[5] / _PI_OVER_180;
-
-    return current_state;
-}
+        /*
+         * Update and bound-check the integral terms.
+         */
+        current_integral += current_error * dt;
+        for(int i=0; i < 6; ++i)
+        {
+            if(fabs(current_integral[i]) > fabs(windup[i]))
+            {
+                current_integral[i] = windup[i] * ((current_integral[i] < 0)?
+                        -1 : 1);
```

Comment:
This is using a standard wrap convention. I'll instead move the current_integral section down.